### PR TITLE
feat: change owners to maintainers

### PR DESCRIPTION
**Pull request by IronCore864**
Per discussions in [this issue](https://github.com/canonical/charm-relation-interfaces/pull/168) (see comment [here](https://github.com/canonical/charm-relation-interfaces/pull/168#issuecomment-2314981570) and [here](https://github.com/canonical/charm-relation-interfaces/pull/168#issuecomment-2316508980)) and an email thread, we have agreed to the followings:

- Use "maintainers" instead of "owners", because "owners" indicates ownership while the interface is meant to be "owned" by a group around it.
- Put a GitHub team ID as "maintainers", instead of a list of users. This eliminates the overhead of maintaining the list of user IDs. The purpose of this field is to assign issues to users when a certain interface test fails; although it's not supported to assign an issue to a team, this can be worked out in the [`charm-relation-interfaces`](https://github.com/canonical/charm-relation-interfaces) repo's [`run_matrix.py`](https://github.com/canonical/charm-relation-interfaces/blob/main/run_matrix.py) script.

This PR to `pytest-interface-tester` reflects the above changes.

See the other part of the change in the [`charm-relation-interfaces`](https://github.com/canonical/charm-relation-interfaces) repo.

An extra note on the version: It has not been used since the last release, 3.1.0. So, although this PR technically is a breaking change, it doesn’t break anything. That’s why I only slightly bumped the version to 3.1.1, not strictly following the SEMVER rules.

### CODE REVIEW

File: interface_tester/collector.py
Comment (PietroPasotti) [inline interface_tester/collector.py:178]:
```suggestion
    maintainers = charms.get("maintainers") or ""
```

so that on the off chance that someone puts a falsy value that is not "" (say, a `null`), we turn it to "" instead

File: interface_tester/collector.py
Comment (PietroPasotti) [inline interface_tester/collector.py:280]: same

**Review by PietroPasotti — APPROVED** [review]
LGTM, modulo a minor consistency nit

**Comment by IronCore864** [comment]
Refactored, please go ahead and merge. Thanks!

File: interface_tester/collector.py
Comment (benhoyt) [inline interface_tester/collector.py:69]: Should this be a `List[str]` as it's maintainers (plural)?

Comment (IronCore864) [reply interface_tester/collector.py:69]: The value will be a single GitHub team ID, not a list of IDs, as in "a team of maintainers". Shall we make it "maintainer"?

Comment (benhoyt) [reply interface_tester/collector.py:69]: Yes, in that case, let's please make it `maintainer` singular.

**Pull request merged**
